mono/metadata/icall.c

```diff
@@ -6104,7 +6104,8 @@ ves_icall_Mono_Runtime_EnableMicrosoftTelemetry (const char *appBundleID, const
 #if defined(TARGET_OSX) && !defined(DISABLE_CRASH_REPORTING)
 	mono_merp_enable (appBundleID, appSignature, appVersion, merpGUIPath, eventType, appPath, configDir);
 
-	mono_get_runtime_callbacks ()->install_state_summarizer ();
+	// Why does this install the sigterm handler so early?
+	// mono_get_runtime_callbacks ()->install_state_summarizer ();
```

Comment:
To avoid race conditions of installing the handler on multiple concurrent crashing threads?
To avoid the heap allocations in the crashing path (see the signal chaining)?

Reply:
Suggest, perhaps, install the handler at the same time, but use a boolean to enable/disable it.